controller_interface/src/controller_interface.cpp

```diff
@@ -65,6 +65,9 @@ const rclcpp_lifecycle::State & ControllerInterface::configure()
       case LifecycleNodeInterface::CallbackReturn::FAILURE:
         break;
     }
+
+    update_rate_ = node_->get_parameter("update_rate").as_int();
```

Comment:
What if we define this on a level of CM and read it together with the `type` of a controller. This would make the interface somewhat simpler. But then we have to add an argument into `update` method.

Reply:
That's a bit of a mess again isn't it? We only keep the controller list under the CM namespace because we don't have a better option... To move it under the CM space we'd need to extend ControllerSpec which you wanted to remove ;)

Reply:
I don't want to remove `ControllerSpec`, it is useful. Maybe it could be a bit restructured.